tests/libs/util/socket_helper.h

```diff
@@ -259,3 +259,20 @@ typedef class _stream_server_socket : public _server_socket
     SOCKET accept_socket;
     size_t message_length;
 } stream_server_socket_t;
+
+/**
+ * @class A helper class for managing WSAStartup and WSACleanup
+ */
+typedef class _wsa_helper
+{
+  public:
+    _wsa_helper();
+    ~_wsa_helper();
+
+    int
+    initialize();
+
+  private:
+    WSADATA data{};
```

Comment:
we don't need this as a member. we don't need to keep it. just have it a local variable in initialize()